src/utils/NovorAdapter.cpp

```diff
@@ -0,0 +1,389 @@
+    os.close();
+
+    // convert mzML to mgf format
+    MzMLFile f;
+    MSExperiment exp;
+    f.setLogType(log_type_);
+    f.load(in, exp);
+ 
+    String tmp_mgf = tmp_dir + "tmp_mgf.mgf";
+    MascotGenericFile mgf;
+    mgf.setLogType(log_type_);
+    mgf.store(tmp_mgf,exp);
+
+    //-------------------------------------------------------------
+    // process
+    //-------------------------------------------------------------
+
+    String tmp_out = tmp_dir + "tmp_out_novor.csv";
+  
+    QStringList process_params;
+    process_params << java_memory
+                   << "-jar" << executable
+                   << "-f" 
+                   << "-o" << tmp_out.toQString()
+                   << "-p" << tmp_param.toQString()
+                   << tmp_mgf.toQString();
+    
+    //TODO: How does ist work with a .bat (Batchfile - Win & sh file linux/mac) 
+    QProcess qp;
+    qp.setWorkingDirectory(path_to_executable);
+    qp.start(java_executable.toQString(), process_params);
+ 
+    // novor command line
+    std::stringstream ss;
+    ss << "COMMAND: " << executable.toStdString();
+    for (QStringList::const_iterator it = process_params.begin(); it != process_params.end(); ++it)
+    {
+        ss << " " << it->toStdString();
+    }
+    LOG_DEBUG << ss.str() << endl;
+
+    // see if process was successfull
+    const bool success = qp.waitForFinished(-1);
+
+    if (success == false || qp.exitStatus() != 0 || qp.exitCode() != 0)
+    {
+      writeLog_( "FATAL: External invocation of Novor failed. Standard output and error were:");
+      const QString nr_stdout(qp.readAllStandardOutput());
+      const QString nr_stderr(qp.readAllStandardError());
+      writeLog_(nr_stdout);
+      writeLog_(nr_stderr);
+      writeLog_(String(qp.exitCode()));
+
+      return EXTERNAL_PROGRAM_ERROR;
+    }
+
+    //-------------------------------------------------------------
+    // writing output
+    //-------------------------------------------------------------
+
+    //TODO: delete tmp dir
+   
+   ifstream file(tmp_out);
+   if (file) 
+   {
+     CsvFile csv(tmp_out, ',');
+          
+     vector<PeptideIdentification> peptide_ids;
+     for (Size i = 0; i != csv.rowCount(); ++i)
+     {
+       StringList sl;
+       csv.getRow(i, sl);
+       
+       if (sl.empty() || sl[0][0] == '#') { continue; }
+       
+       PeptideIdentification pi;
+       pi.setMetaValue("scan_index", sl[1].toDouble());
+       pi.setScoreType("novorscore");
+       pi.setHigherScoreBetter(true);
+       pi.setRT(sl[2].toDouble());
+       pi.setMZ(sl[3].toDouble());
+
+       PeptideHit ph;
+       ph.setCharge(sl[4].toInt());
+       ph.setScore(sl[8].toDouble());
+
+       // replace PTM name (see http://wiki.rapidnovor.com/wiki/Built-in_PTMs)
+       String sequence = sl[9];
+       sequence.substitute("(Cam)", "(Carbamidomethyl)");
+       sequence.substitute("(O)","(Oxidation)");
+       sequence.substitute("(PyroCam)", "(Pyro-carbamidomethyl)");
```

Comment:
the list on the wiki is substantially longer than the ones here, will this be a problem if there are mods in the output file that are not recognized here?

Reply:
l think these are the only ones which have a different "short name" compared to unimod, so there should not be any problems with the others. If the list of Novor-modifications is extended we may have to check again and update. Will add a comment in the code - or do you have another suggestion in that regard.